File: ccif/src/java/other/org/himalay/ccif/HdmiStatusReport.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:56 EST 2016")

public  class HdmiStatusReport extends BinStruct implements PublicBinMsg{




    // members variables
    // bf1
    public BitField_8 bf1 ;
    // vdoFormat
    public VideoFormat vdoFormat ;
    // adoFormat
    public AudioFormat adoFormat ;

        public HdmiStatusReport () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize bf1
        bf1 = new BitField_8();
        // Initialize vdoFormat
        vdoFormat = new VideoFormat();
        // Initialize adoFormat
        adoFormat = new AudioFormat();
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read bf1
        retVal += bf1.read(istream);
        // read vdoFormat
        retVal += vdoFormat.read(istream);
        // read adoFormat
        retVal += adoFormat.read(istream);

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        { /** fix dependent sizes for vdoFormat **/  }
        { /** fix dependent sizes for adoFormat **/  }
    
                // write bf1
        ostream.writeByte(bf1.getValue()); retVal +=1;
        // write vdoFormat
        if (vdoFormat!=null)retVal +=vdoFormat.write(ostream);
        // write adoFormat
        if (adoFormat!=null)retVal +=adoFormat.write(ostream);
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("HdmiStatusReport\n");
    dc.increaseIndent();
        int retVal= 0;
        // write bf1
        {dc.indent();dc.getPs().print("deviceType: ");dc.getPs().println(BitField_8.toDisplayString(getDeviceType(),1));dc.indent();dc.getPs().print("colorSpace: ");dc.getPs().println(BitField_8.toDisplayString(getColorSpace(),2));dc.indent();dc.getPs().print("connectionStatus: ");dc.getPs().println(BitField_8.toDisplayString(getConnectionStatus(),2));dc.indent();dc.getPs().print("HostHdcpStatus: ");dc.getPs().println(BitField_8.toDisplayString(getHostHdcpStatus(),1));dc.indent();dc.getPs().print("deviceHdcpStatus: ");dc.getPs().println(BitField_8.toDisplayString(getDeviceHdcpStatus(),2));}
        // write vdoFormat
        if ( vdoFormat != null ) {dc.indent();dc.getPs().println("vdoFormat") ;retVal +=vdoFormat.dump(dc);}
        // write adoFormat
        if ( adoFormat != null ) {dc.indent();dc.getPs().println("adoFormat") ;retVal +=adoFormat.dump(dc);}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for bf1
    //public BitField_8 getBf1()
    //{
    //    return bf1 ;
    //}

    
    // Setter for bf1
    //public void setBf1(BitField_8 val)
    //{
    //    this.bf1= val;
    //}
    // Getter for vdoFormat
    //public VideoFormat getVdoFormat()
    //{
    //    return vdoFormat ;
    //}

    
    // Setter for vdoFormat
    //public void setVdoFormat(VideoFormat val)
    //{
    //    this.vdoFormat= val;
    //}
    // Getter for adoFormat
    //public AudioFormat getAdoFormat()
    //{
    //    return adoFormat ;
    //}

    
    // Setter for adoFormat
    //public void setAdoFormat(AudioFormat val)
    //{
    //    this.adoFormat= val;
    //}


    public int getDeviceType( )
    {
        return ( bf1.getValue() & 0x00000080 ) >> 7 ;
    }
    
    
    public void setDeviceType(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffff7f) | ( (val << 7 ) & 0x00000080));
    }
    
    
    public int getColorSpace( )
    {
        return ( bf1.getValue() & 0x00000060 ) >> 5 ;
    }
    
    
    public void setColorSpace(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffff9f) | ( (val << 5 ) & 0x00000060));
    }
    
    
    public int getConnectionStatus( )
    {
        return ( bf1.getValue() & 0x00000018 ) >> 3 ;
    }
    
    
    public void setConnectionStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffffe7) | ( (val << 3 ) & 0x00000018));
    }
    
    
    public int getHostHdcpStatus( )
    {
        return ( bf1.getValue() & 0x00000004 ) >> 2 ;
    }
    
    
    public void setHostHdcpStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffffffb) | ( (val << 2 ) & 0x00000004));
    }
    
    
    public int getDeviceHdcpStatus( )
    {
        return ( bf1.getValue() & 0x00000003 ) >> 0 ;
    }
    
    
    public void setDeviceHdcpStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffffffc) | ( (val << 0 ) & 0x00000003));
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/HdmiStatusReport.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:14 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "HdmiStatusReport.h"

#ifdef org_himalay_ccif_HdmiStatusReport__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
HdmiStatusReport::HdmiStatusReport() {
    references= 0;


    // bf1
    bf1 = 0;
    // vdoFormat
    org_himalay_ccif_VideoFormat__NEW( vdoFormat);
    // adoFormat
    org_himalay_ccif_AudioFormat__NEW( adoFormat);
}
// Destructor
HdmiStatusReport::~HdmiStatusReport() {
    // bf1

    // vdoFormat
    org_himalay_ccif_VideoFormat__DELETE( vdoFormat);
    // adoFormat
    org_himalay_ccif_AudioFormat__DELETE( adoFormat);
}

int HdmiStatusReport::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // bf1
    bf1= istream.readUI8();
    retVal += 1;
    // vdoFormat
    retVal += vdoFormat->read(istream);
    // adoFormat
    retVal += adoFormat->read(istream);
    return retVal;
}

int HdmiStatusReport::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read bf1
    bf1= istream.readUI8();
    retVal += 1;
    // read vdoFormat
    retVal += vdoFormat->read(istream);
    // read adoFormat
    retVal += adoFormat->read(istream);

    return retVal;
}

int HdmiStatusReport::write(DataOutputStream& ostream) { // throws IOException

    {   /** fix dependent sizes for vdoFormat **/
    }
    {   /** fix dependent sizes for adoFormat **/
    }

    int retVal= 0;
    // write bf1
    ostream.writeUI8(bf1);
    retVal +=1;
    // write vdoFormat
    if ( vdoFormat != NULL ) {
        retVal +=vdoFormat->write(ostream);
    }
    // write adoFormat
    if ( adoFormat != NULL ) {
        retVal +=adoFormat->write(ostream);
    }

    return retVal;
}

int HdmiStatusReport::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("HdmiStatusReport\n");
    dc.increaseIndent();
    int retVal= 0;
    // write bf1
    {
        dc.indent();
        dc.getPs().print("deviceType: ");
        dc.getPs().printlnBin(getDeviceType(),1);
        dc.indent();
        dc.getPs().print("colorSpace: ");
        dc.getPs().printlnBin(getColorSpace(),2);
        dc.indent();
        dc.getPs().print("connectionStatus: ");
        dc.getPs().printlnBin(getConnectionStatus(),2);
        dc.indent();
        dc.getPs().print("HostHdcpStatus: ");
        dc.getPs().printlnBin(getHostHdcpStatus(),1);
        dc.indent();
        dc.getPs().print("deviceHdcpStatus: ");
        dc.getPs().printlnBin(getDeviceHdcpStatus(),2);
    }
    // write vdoFormat
    if ( vdoFormat != NULL ) {
        dc.indent();
        dc.getPs().println("vdoFormat") ;
        retVal +=vdoFormat->dump(dc);
    }
    // write adoFormat
    if ( adoFormat != NULL ) {
        dc.indent();
        dc.getPs().println("adoFormat") ;
        retVal +=adoFormat->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for bf1
//ui8 HdmiStatusReport::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void HdmiStatusReport::setBf1(ui8 val)
//{
//this->bf1= val;
//}

// Getter for vdoFormat
//org_himalay_ccif_VideoFormat__PTR_TYPE HdmiStatusReport::getVdoFormat()
//{
//return this->vdoFormat;
//}

// Setter for vdoFormat
//void HdmiStatusReport::setVdoFormat(org_himalay_ccif_VideoFormat__PTR_TYPE val)
//{
//this->vdoFormat= val;
//}

// Getter for adoFormat
//org_himalay_ccif_AudioFormat__PTR_TYPE HdmiStatusReport::getAdoFormat()
//{
//return this->adoFormat;
//}

// Setter for adoFormat
//void HdmiStatusReport::setAdoFormat(org_himalay_ccif_AudioFormat__PTR_TYPE val)
//{
//this->adoFormat= val;
//}

int HdmiStatusReport::getDeviceType( ) {
    return ( this->bf1 & 0x00000080 ) >> 7 ;
}

void HdmiStatusReport::setDeviceType(int val) {
    bf1=(  (this->bf1 & 0xffffff7f) | ( (val << 7 ) & 0x00000080));
}

int HdmiStatusReport::getColorSpace( ) {
    return ( this->bf1 & 0x00000060 ) >> 5 ;
}

void HdmiStatusReport::setColorSpace(int val) {
    bf1=(  (this->bf1 & 0xffffff9f) | ( (val << 5 ) & 0x00000060));
}

int HdmiStatusReport::getConnectionStatus( ) {
    return ( this->bf1 & 0x00000018 ) >> 3 ;
}

void HdmiStatusReport::setConnectionStatus(int val) {
    bf1=(  (this->bf1 & 0xffffffe7) | ( (val << 3 ) & 0x00000018));
}

int HdmiStatusReport::getHostHdcpStatus( ) {
    return ( this->bf1 & 0x00000004 ) >> 2 ;
}

void HdmiStatusReport::setHostHdcpStatus(int val) {
    bf1=(  (this->bf1 & 0xfffffffb) | ( (val << 2 ) & 0x00000004));
}

int HdmiStatusReport::getDeviceHdcpStatus( ) {
    return ( this->bf1 & 0x00000003 ) >> 0 ;
}

void HdmiStatusReport::setDeviceHdcpStatus(int val) {
    bf1=(  (this->bf1 & 0xfffffffc) | ( (val << 0 ) & 0x00000003));
}

int HdmiStatusReport::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_HdmiStatusReport__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::HdmiStatusReport* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::HdmiStatusReport* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code